src/attack.rs

```diff
@@ -47,28 +49,440 @@ pub struct Attack {
+//
+//                     let mut offset = Vec2::new(24.0, 0.0);
+//                     let mut attack_frames = AttackFrames {
+//                         startup: 0,
+//                         active: 0,
+//                         recovery: 0,
+//                     };
+//
+//                     if let Some(fighter) = fighter_assets.get(fighter_handle) {
+//                         offset = fighter.attack.hitbox_offset;
+//                         attack_frames = fighter.attack.frames;
+//
+//                     } else {
+//                         println!("not loading from assets?")
+//                     };
+//
+//                     let direction = match facing {
+//                         Facing::Left => -1.0,
+//                         Facing::Right => 1.0,
+//                     };
+//                     offset *= direction;
+//
+//                     let attack_entity = commands
+//                         .spawn_bundle(TransformBundle::from_transform(
+//                             Transform::from_translation(offset.extend(0.0)),
+//                         ))
+//                         .insert(Sensor)
+//                         .insert(ActiveEvents::COLLISION_EVENTS)
+//                         .insert(
+//                             ActiveCollisionTypes::default() | ActiveCollisionTypes::STATIC_STATIC,
+//                         )
+//                         .insert(CollisionGroups::new(
+//                             BodyLayers::ENEMY_ATTACK,
+//                             BodyLayers::PLAYER,
+//                         ))
+//                         .insert(Attack { damage: 10 })
+//                         .insert(attack_frames)
+//                         // .insert(fighter)
+//                         // .insert(AttackFrames {
+//                         //     startup: 1,
+//                         //     active: 2,
+//                         //     recovery: 3,
+//                         // })
+//                         .id();
+//                     commands.entity(event.0).push_children(&[attack_entity]);
+//
+//                     if let Some(fighter) = fighter_assets.get(fighter_handle) {
+//                         if let Some(effects) = fighter.audio.effect_handles.get(&state) {
+//                             let fx_playback =
+//                                 FighterStateEffectsPlayback::new(*state, effects.clone());
+//                             commands.entity(entity).insert(fx_playback);
+//                         }
+//                     }
+//                 }
+//             }
+//         }
+//     }
+// }
+
 fn activate_hitbox(
     attack_query: Query<(Entity, &AttackFrames, &Parent), Without<Collider>>,
-    animated_query: Query<&Animation>,
+    fighter_query: Query<(&Animation, &Handle<FighterMeta>)>,
     mut commands: Commands,
+    //maybe change to using attack meta? or atatcks defined in their own files but still loaded via
```

Comment:
Maybe...but I'd probably wait until we have something other than fighters that actually wants to use this attack system.